Packs/Illumio/Integrations/IllumioCore/IllumioCore.py

```diff
@@ -22,6 +28,171 @@ def __init__(self, port):
+        protocol: Communication protocol.
+
+    Returns:
+        str: Error if value of the argument is not valid.
+    """
+    if port < MIN_PORT or port > MAX_PORT:  # type: ignore
+        raise InvalidPortException(port)
+
+    for decision in policy_decisions:
+        if decision not in VALID_POLICY_DECISIONS:
+            raise InvalidMultiSelectException('policy_decisions', VALID_POLICY_DECISIONS)
+
+    if protocol not in VALID_PROTOCOLS:
+        raise InvalidSingleSelectException(protocol, "protocol", VALID_PROTOCOLS)
+
+
+def validate_arguments_for_illumio_virtual_service_create_command(name: str | None, port: str | None,
+                                                                  protocol: str) -> tuple:
+    """Validate and convert arguments for illumio virtual service create command.
+
+    Args:
+        name: Name of virtual service.
+        port: Port number.
+        protocol: Protocol name.
+
+    Returns:
+        Returns tuple of converted values (port, protocol).
+    """
+    validate_required_parameters(name=name, port=port)
+
+    port = arg_to_number(port, arg_name="port")  # type: ignore
+
+    if port != -1 and (port > MAX_PORT or port < MIN_PORT):  # type: ignore
+        raise ValueError("{} is an invalid value for port. Value must be in 1 to 65535 or -1.".format(port))
+
+    if protocol not in VALID_PROTOCOLS:
+        raise InvalidSingleSelectException(protocol, 'protocol', VALID_PROTOCOLS)
+
+    proto = convert_protocol(protocol)
+    return port, proto
+
+
+
+def prepare_hr_for_traffic_analysis(response: list, protocol: str) -> str:
+    """Prepare Human Readable output for traffic analysis command.
+
+    Args:
+        response: Response from the SDK.
+        protocol: Communication protocol.
+
+    Returns:
+        markdown string to be displayed in the war room.
+    """
+    hr_output = []
+    for traffic in response:
+        hr_output.append({
+            "Source IP": traffic.get('src', {}).get('ip'),
+            "Destination IP": traffic.get('dst', {}).get('ip'),
+            "Destination Workload Hostname": traffic.get('dst', {}).get('workload', {}).get('hostname'),
+            "Service Port": traffic.get('service', {}).get('port'),
+            "Service Protocol": protocol,
+            "Policy Decision": traffic.get('policy_decision'),
+            "State": traffic.get("state"),
+            "Flow Direction": traffic.get('flow_direction'),
+            "First Detected": arg_to_datetime(traffic.get('timestamp_range', {}).get('first_detected', '')).strftime(  # type: ignore # noqa
+                HR_DATE_FORMAT),  # type: ignore # noqa
+            "Last Detected": arg_to_datetime(traffic.get('timestamp_range', {}).get('last_detected', '')).strftime(  # type: ignore # noqa
+                HR_DATE_FORMAT)
+        })
+
+    headers = ["Source IP", "Destination IP", "Destination Workload Hostname", "Service Port", "Service Protocol",
```

Comment:
These hard-coded header arrays are unnecessary. In the worst case you could simply capture the dict `keys()` above, but I think the parameter can be omitted entirely if the header values match the keys.

Reply:
We cannot omit the parameter as we're using the commonserverpython method, but we can directly capture the `keys ` instead

Reply:
See [this example](https://github.com/demisto/content/blob/master/Packs/AWS-EC2/Integrations/AWS-EC2/AWS-EC2.py#L103). I'm pretty sure based on usage in other packs that the default behaviour if no headers list is passed is to use the keys as-is.

Reply:
Yes, will remove the headers parameter

Reply:
@dsommerville-illumio, we cannot remove this parameter as the `table_to_markdown` function will prepare the table headers in a random sequence & it'll not match with the sequence mentioned in TDD, so we need to pass headers explicitly.
We have directly captured the `keys `instead of hardcoding them and passed them in the `table_to_markdown` function.

Reply:
If the sequence is important, would an OrderedDict work here?

Reply:
Yes, we'll use OrderedDict.